### DISCUSSION TITLE
Apache is Working But MySQL Hangs/Freezes on Initializing Data

### DISCUSSION THREAD

**Discussion by chytons** [leokhoa/laragon#797] [Q&A] [answered]
Good day,
I installed Laragon v6 and having trouble starting MySQL server.
When I click on Start All, Apache starts immediately but MySQL freezes on Initializing data.
I have installed in the default location and also other locations without success.
Below are the details of my environment:

OS - Windows 10 22H2 Build 19045.4474
RAM - 32 GB

Funny enough, I have another PC with similar configuration and everything is working fine on it.

![laragon-screen-capture](https://github.com/leokhoa/laragon/assets/53003784/06ccafde-352a-4fd1-a805-edf08826200c)

**Comment by Pen-y-Fan** [comment]
The first thing I would check is permissions see my blog [How to set up Laragon on a new Windows computer (part 2) - permissions](https://pen-y-fan.github.io/2022/11/08/how-to-set-up-laragon-on-a-new-windows-computer-part-2/)

**Reply by chytons** [reply]
I followed your instructions as stated in the tutorial, but the issue still persist.
I uninstalled and installed again and the issue is still there.
Attached is the log file.
[mysql_error.log.txt](https://github.com/user-attachments/files/15594070/mysql_error.log.txt)

**Reply by Pen-y-Fan** [reply]
Try deleting all the files in C:\laragon\data\mysql-8\

See if you can create a text file in that directory.

**Comment by chytons** [comment]
I have deleted all the files and I am able to create and empty text file inside the mysql-8 folder.
![laragon-data-folder](https://github.com/leokhoa/laragon/assets/53003784/587f5a48-8a23-4164-bff5-ef49956921f0)

**Reply by Pen-y-Fan** [reply]
Check for Windows updates, complete any updates and reboot the PC.

**Reply by chytons** [reply]
My Windows OS is up to date with the latest build of Windows 10 22H2 Build 19045.4474 as I communicated earlier.
I have restarted the system multiple times. The issue still persist.
![Windows OS](https://github.com/leokhoa/laragon/assets/53003784/fadf6912-3ccd-44d1-a216-313da1760dc3)

**Reply by Pen-y-Fan** [reply]
Ok, it's not permissions or an update. Next, check anti-virus logs for false positives or things being blocked or deleted.

**Comment by chytons** [comment]
I did disable the anti-virus.

I have portable version of XAMPP on my system. I start/stop Apache and MySQL manually. Neither XAMPP's Apache nor MySQL was running while troubleshooting Laragon issue. I have changed the folder path where XAMPP is installed in case of conflict. I have completely removed XAMPP using Revo UninstallerPro tool, but a reference is being made to the delete location of XAMPP installation. I have also changed MySQL port to 3406. I have given full control permission on C:\laragon to Authenticated Users and Users.
I set up two Windows 10 VMs and laragon installations in them are working. I copied the mysql-8 folder from one of the VM into the same location of my system. MySQL started running but exited with error after some seconds. I have attached all the relevant screen captures here.

[mysqld.log](https://github.com/user-attachments/files/15677111/mysqld.log)

![laragon-mysql-error](https://github.com/leokhoa/laragon/assets/53003784/3f46b79d-0490-4b49-9d5f-dcd364140543)


https://github.com/leokhoa/laragon/assets/53003784/b12fe896-cbdb-4532-bc74-34ea1cf2edf6

**Reply by Pen-y-Fan** [reply]
The error message is different, this is usually progress! 

There is some confusion over the directory being used, c:\Servers\xampp\tmp in the screenshot above, C:\Servers\laragon\data\mysql-8\ in the original screenshot (I mistook this for c:\laragon\data\mysql-8\)

When copying Laragon from one system to another a config file must be hard-coded with a data directory, which doesn't exist in the new system.

I recommend starting from a clean VM and installing Laragon using the installer. Once installed and running, migrate the data using MySQL dump and import using mysql.

**Reply by chytons** [reply, marked as answer]
After going through the steps, it occurred to me that I have an old my.cnf file directly on my C:\ drive which was pointing to XAMPP installation path. This was always read every time I started MySQL server.
After renaming the file, the issue disappeared like magic.

Thanks for all the assistance.

**Comment by saakiiib** [comment]
I'm facing same issue. But unfortunately I dont find any file named my.cnf anywhere. @chytons  
![image](https://github.com/leokhoa/laragon/assets/128025804/05a8f455-84b7-4b74-96a4-92a6c2ede875)

**Reply by chytons** [reply]
@saakiiib, did you previously install any LAMP stack i.e. XAMPP, WAMP etc.?

**Reply by saakiiib** [reply]
@chytons Yes currently my XAMPP is installed on my machine and I renamed those my.ini files. But still same issue I tried third party apps for searching. Do you have any suggestions?

**Comment by chytons** [comment]
Checkout the link below for further troubleshooting.

https://github.com/leokhoa/laragon/issues/798#issuecomment-2153334273
